### ISSUE TITLE
Scalar type VS bit-fiddling

### ISSUE THREAD

**Issue by oleganza** [isislovecruft/x25519-dalek#1]
I've noticed that `curve25519_dalek::Scalar` is used as a bag of bytes to store both random 32-bytes (when you use `diffie_hellman()` function) and fiddled 32-bytes (the output of `decode_scalar()`). In both cases the type `Scalar` does not store a "proper" scalar modulo curve order.

While technically it works, this makes me feel uneasy. E.g. what if `curve25519_dalek` library later (if not already) assumes that its `Scalar` type is always properly reduced, and I use the same scalar with x25519 library and with some custom crypto protocol? Does it make sense maybe to introduce a x25519-specific "SecretKey" type or something like that, which does not pretend to be a scalar mod L? Or maybe I'm overthinking this and it's fine as is?

**Comment by isislovecruft** [comment]
I agree with adding a SecretKey type, and I've added this to the checklist in #9. Perhaps we could provide a `impl From<x25519_dalek::SecretKey> for curve25519_dalek::scalar::Scalar`? Although.. I'm not really sure why anyone would do that versus just doing `let shared = diffie_hellman(my_secret, their_public); let s = Scalar::from_bytes_mod_order(shared);`?
